app/controllers/incomeSources/add/IncomeSourceCheckDetailsController.scala

```diff
@@ -0,0 +1,229 @@
+              businessPostalCode = address.postcode,
+              businessCountryCode = addIncomeSourceData.countryCode,
+              incomeSourcesAccountingMethod = addIncomeSourceData.incomeSourcesAccountingMethod,
+              cashOrAccrualsFlag = addIncomeSourceData.incomeSourcesAccountingMethod
+                .getOrElse(throw MissingSessionKey(s"$errorTracePrefix incomeSourcesAccountingMethod")),
+              showedAccountingMethod = showAccountingMethodPage,
+              incomeSourceType = SelfEmployment
+            ))
+
+          case None => throw new Exception(s"$errorTracePrefix failed to retrieve addIncomeSourceData")
+        }
+      case _ => throw new Exception(s"$errorTracePrefix failed to retrieve uiJourneySessionData ")
+    }
+
+  }
+
+  def submit(incomeSourceType: IncomeSourceType): Action[AnyContent] = (checkSessionTimeout andThen authenticate andThen retrieveNino
+    andThen retrieveIncomeSources andThen retrieveBtaNavBar).async {
+    implicit user =>
+      handleSubmit(isAgent = false, incomeSourceType)
+  }
+
+  def submitAgent(incomeSourceType: IncomeSourceType): Action[AnyContent] = Authenticated.async {
+    implicit request =>
+      implicit user =>
+        getMtdItUserWithIncomeSources(incomeSourceDetailsService).flatMap {
+          implicit mtdItUser =>
+            handleSubmit(isAgent = true, incomeSourceType)
+        }
+  }
+
+  private def handleSubmit(isAgent: Boolean, incomeSourceType: IncomeSourceType)(implicit user: MtdItUser[_]): Future[Result] = withIncomeSourcesFS {
+    val (redirect, errorRedirect) = (isAgent, incomeSourceType) match {
+      case (true, SelfEmployment) => (routes.BusinessReportingMethodController.showAgent _, routes.IncomeSourceNotAddedController.showAgent(SelfEmployment).url)
+      case (false, SelfEmployment) => (routes.BusinessReportingMethodController.show _, routes.IncomeSourceNotAddedController.show(SelfEmployment).url)
+      case (true, UkProperty) => (routes.UKPropertyReportingMethodController.showAgent _, routes.IncomeSourceNotAddedController.showAgent(UkProperty).url)
+      case (false, UkProperty) => (routes.UKPropertyReportingMethodController.show _, routes.IncomeSourceNotAddedController.show(UkProperty).url)
+      case (true, ForeignProperty) => (routes.ForeignPropertyReportingMethodController.showAgent _, routes.IncomeSourceNotAddedController.showAgent(ForeignProperty).url)
+      case (false, ForeignProperty) => (routes.ForeignPropertyReportingMethodController.show _, routes.IncomeSourceNotAddedController.show(ForeignProperty).url)
+    }
+    {
+      incomeSourceType match {
+        case SelfEmployment => getBusinessModel
+        case _ => getPropertyModel(incomeSourceType)
+      }
+    }.flatMap {
+      case Right(viewModel) =>
+        businessDetailsService.createRequest(viewModel).flatMap {
+          case Right(CreateIncomeSourceResponse(id)) =>
+            sessionService.deleteMongoData(JourneyType(Add, incomeSourceType))
```

Comment:
* this would not work as expected, as you "cutting off" Future to be completed: line => deleteMongoData